pkg/client/cache/reflector.go

```diff
@@ -78,7 +84,32 @@ func NewNamespaceKeyedIndexerAndReflector(lw ListerWatcher, expectedType interfa
 // so that you can use reflectors to periodically process everything as well as
 // incrementally processing the things that change.
 func NewReflector(lw ListerWatcher, expectedType interface{}, store Store, resyncPeriod time.Duration) *Reflector {
+	// walk back callers until we find a caller from outside of the "cache" package
+	name := "????"
```

Comment:
This belongs someplace else (utility method) that lets you specify all these things as arguments.